### DISCUSSION TITLE
What are the limitations of bind mounts?

### DISCUSSION THREAD

**Discussion by skellycode** [Jip-Hop/jailmaker#53] [Q&A] [answered]
Hi!

In the documentation, it specifies:
"Do not bind your TrueNAS system directories (/root /mnt..."
It also states:
"E.g creating a websites dataset in a pool named tank then binding --bind='/mnt/tank/websites/websitename/:/var/www/websitename/'"

So my question is two-fold.
Firstly, is binding `/mnt/tank`, or `/mnt/Apps` (or whatever the actual pool is called) ok?  This is not `/mnt`, however it is still a root-level pool.

Secondly, can we bind datasets?  In the example above, it states creating a websites dataset, but then proceeds to bind a subdirectory of that dataset.  Does this mean mounting a dataset directly is not recommended?

**Comment by Jip-Hop** [comment]
It should all be fine. You can essentially mount whatever you want. Just be careful not to mess op the host OS. It should be no problem to mount a dataset directly.

**Reply by skellycode** [reply]
Can you please give an example of 'be careful not to mess up the host OS'.  How does one mess it up? (and avoid it, for that matter!).  It's not clear to me what is a good or bad idea 🤷

**Reply by Jip-Hop** [reply, marked as answer]
What would be a bad idea is to mount the /bin directory somewhere in the jail and then remove all files inside it... As long as you mount directories you've made yourself on your own datasets (including the dataset itself) you should be fine.

**Reply by skellycode** [reply]
That is much clearer.  Thank you 😄

**Comment by spusuf** [comment]
For TrueNAS to effectively manage it's data it manages datasets with permissions in software for each dataset. There is no write protections on the host OS (located on your boot drive) so if you were to bind /bin, /etc, /var or any of those other host locations with TrueNAS software you risk allowing the jail to overwrite the system files and break your install. You technically CAN bind the pools (groups of drives) but it is highly discouraged due to breaking permissions for any other use of the space in TrueNAS. Correct way is to make a dataset inside the pool and mount that or mount a subdirectory. 

TLDR: /mnt contains your pools, your pools contain your datasets. Use datasets. If you /mnt or anything above it you risk overwriting TrueNAS files.
